docs/developer-guide/extensions/ui-extensions.md

```diff
@@ -158,3 +161,65 @@ Below is an example of an extension using the flyout widget:
   );
 })(window);
 ```
+
+## Top Bar Action Menu Extensions
+
+The top bar panel is the action menu at the top of the application view where the action buttons are displayed like Details, Sync, Refresh. Argo CD allows you to add new button to the top bar action menu of an application.
+When the extension button is clicked, the custom widget will be rendered in a flyout panel.
+
+The extension should be registered using the `extensionsAPI.registerTopBarActionMenuExt` method:
+
+```typescript
+registerTopBarActionMenuExt(
+        component: TopBarActionMenuExtComponent,
+        title: string,
+        id: string,
+        flyout?: ExtensionComponent,
+        shouldDisplay: (app?: Application) => boolean = () => true,
+        iconClassName?: string,
+        isMiddle = false
+)
+```
+
+The callback function `shouldDisplay` should return true if the extension should be displayed and false otherwise:
+
+```typescript
+  const shouldDisplay =  (app: Application) => {
+    return application.metadata?.labels?.['application.environmentLabelKey'] === "prd";
+  };
+  const flyout = () => {
+    return React.createElement(
+            "div",
+            { style: { padding: "10px" } },
+            "This is a flyout"
+    );
+  };
+  const component = () => {
+    return React.createElement(
+            "div",
+            {
+              onClick: () => flyout()
+            },
+            "Toolbar Extension Test"
+    );
+  };
```

Comment:
I'd move the `flyout` and `component` into the example below as they are examples and pure JS. I would also keep the formatting identical to the prior examples

whereas the `shouldDisplay` part above this can remain in TS as it shows a type signature, same as the registration function